agent/exec/container/executor.go

```diff
@@ -41,12 +41,35 @@ func (e *executor) Describe(ctx context.Context) (*api.NodeDescription, error) {
 		}
 	}
 
+	// add v1 plugins to 'plugins'
 	addPlugins("Volume", info.Plugins.Volume)
 	// Add builtin driver "overlay" (the only builtin multi-host driver) to
 	// the plugin list by default.
 	addPlugins("Network", append([]string{"overlay"}, info.Plugins.Network...))
 	addPlugins("Authorization", info.Plugins.Authorization)
 
+	// retrieve v2 plugins
+	v2plugins, err := e.client.PluginList(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// add v2 plugins to 'plugins'
+	for _, plgn := range v2plugins {
+		for _, typ := range plgn.Config.Interface.Types {
+			plgnTyp := typ.Capability
+			if typ.Capability == "volumedriver" {
+				plgnTyp = "Volume"
```

Comment:
@tiborvass Not exactly. The way I understand, the reason is in `filter.go`. Notice the `Check()` function for `PluginFilter`

```
// Check returns true if the task can be scheduled into the given node.
func (f *PluginFilter) Check(n *NodeInfo) bool {
	// Get list of plugins on the node
	nodePlugins := n.Description.Engine.Plugins

	// Check if all volume plugins required by task are installed on node
	container := f.t.Spec.GetContainer()
	if container != nil {
		for _, mount := range container.Mounts {
			if mount.VolumeOptions != nil && mount.VolumeOptions.DriverConfig != nil {
				if !f.pluginExistsOnNode("Volume", mount.VolumeOptions.DriverConfig.Name, nodePlugins) {
					return false
				}
			}
		}
	}

	// Check if all network plugins required by task are installed on node
	for _, tn := range f.t.Networks {
		if !f.pluginExistsOnNode("Network", tn.Network.DriverState.Name, nodePlugins) {
			return false
		}
	}
	return true
}
```
In particular, I'm pointing out to the fact that volumes are stored as mounts, and networks stored separately, and we check if the corresponding Driver exists on that node. This is why I wanted to store with a simple "Volume" or "Network" key, instead of the longer form returned by `type.String()`. For other plugins, this PR is storing the long form version.